RadixWallet/Features/CreateAccount/Coordinator/CreateAccountCoordinator+Reducer.swift

```diff
@@ -112,33 +145,137 @@ extension CreateAccountCoordinator {
 
 	public func reduce(into state: inout State, childAction: ChildAction) -> Effect<Action> {
 		switch childAction {
-		case let .root(.step1_nameAccount(.delegate(.proceed(accountName, useLedgerAsFactorSource)))):
-			state.path.append(.step2_creationOfAccount(.init(
-				name: accountName,
-				networkID: state.config.specificNetworkID,
-				isCreatingLedgerAccount: useLedgerAsFactorSource
-			)))
+		case let .root(.nameAccount(.delegate(.proceed(accountName, useLedgerAsFactorSource)))):
+			state.name = accountName
+			if useLedgerAsFactorSource {
+				state.path.append(.selectLedger(.init(context: .createHardwareAccount)))
+				return .none
+			} else {
+				return .send(.internal(.derivePublicKeyFromDevice))
+			}
+
+		case let .path(.element(_, action: .selectLedger(.delegate(.choseLedger(ledger))))):
+			return .send(.internal(.derivePublicKeyFromLedger(ledger)))
+
+		case .path(.element(_, action: .completion(.delegate(.completed)))):
+			return .run { send in
+				await send(.delegate(.completed))
+				if isPresented {
+					await dismiss()
+				}
+			}
+
+		default:
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case .derivePublicKeyFromDevice:
+			state.destination = .derivePublicKey(
+				.init(
+					derivationPathOption: .next(
+						for: .account,
+						networkID: state.config.specificNetworkID,
+						curve: .curve25519,
+						scheme: .cap26
+					),
+					factorSourceOption: .device,
+					purpose: .createNewEntity(kind: .account)
+				))
+			return .none
+
+		case let .derivePublicKeyFromLedger(ledger):
+			state.destination = .derivePublicKey(
+				.init(
+					derivationPathOption: .next(
+						for: .account,
+						networkID: state.config.specificNetworkID,
+						curve: .curve25519,
+						scheme: .cap26
+					),
+					factorSourceOption: .specific(ledger.embed()),
+					purpose: .createNewEntity(kind: .account)
+				))
 			return .none
 
-		case let .path(.element(_, action: .step2_creationOfAccount(.delegate(.createdAccount(newAccount))))):
-			state.path.append(.step3_completion(.init(
-				account: newAccount,
+		case let .createAccountResult(.success(account)):
+			return .run { send in
+				try await accountsClient.saveVirtualAccount(account)
+				await send(.internal(.handleAccountCreated(account)))
+			} catch: { error, send in
+				loggerGlobal.error("Failed to save newly created virtual account into profile: \(error)")
+				await send(.internal(.handleFailure))
```

Comment:
could have the same pattern as for `createAccountResult`? instead of separate actions for success and failure.

Reply:
Not following you in this case. You would want to manually send `await send(.internal(.createAccountResult(.failure(error))))` instead?

Reply:
I mean for consistency with `createAccountResult`  instead of having in Internal action enum:
```
case handleAccountCreated(Profile.Network.Account)
case handleFailure
```
you could have:
```
case handleAccountCreated(TaskResult<Profile.Network.Account>)
```

then here you could have:
```
return .run { send in
  let result = TaskResult { try await accountsClient.saveVirtualAccount(account) }
  await send(.internal(.handleAccountCreated(result)))
}
```